src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -12,37 +13,60 @@
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes a person identified using it's displayed index or displayed name from the address book.
  */
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person identified by the index number or full name used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer) or NAME (must be the exact full name)\n"
+            + "Example: " + COMMAND_WORD + " 1" + "or" + COMMAND_WORD + " John Doe";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
     private final Index targetIndex;
-
-    public DeleteCommand(Index targetIndex) {
+    private final String targetName;
+
+    /**
+     * Constructs a DeleteCommand with the specified target index and target name.
+     *
+     * @param targetIndex The index of the person to be deleted.
+     * @param targetName The name of the person to be deleted.
+     */
+    public DeleteCommand(Index targetIndex, String targetName) {
         this.targetIndex = targetIndex;
+        this.targetName = targetName;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        String dummyName = " ";
+
+        if (targetIndex != null) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deletePerson(personToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
+
+        } else if (targetName != dummyName) {
+            Optional<Person> personToFind = lastShownList.stream().filter(person -> person.getName()
+                    .toString().equals(targetName)).findFirst();
+            if (personToFind.isPresent()) {
+                Person personToDelete = personToFind.get();
+                model.deletePerson(personToDelete);
+                return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
+            } else {
+                throw new CommandException("Person with the name \"" + targetName + "\" not found.");
```

Comment:
For greater consistency, would it be better to abstract the String and declare it at the top of the file as a static final variable, similar to MESSAGE_DELETE_PERSON_SUCCESS?